ui/src/services/alertmanager/api.js

```diff
@@ -1,11 +1,35 @@
+//@flow
 import ApiClient from '../ApiClient';
 
-let alertmanagerApiClient = null;
+let alertmanagerApiClient: ApiClient = null;
 
-export function initialize(apiUrl) {
+export function initialize(apiUrl: string) {
   alertmanagerApiClient = new ApiClient({ apiUrl });
 }
 
-export function getAlerts() {
+export type PrometheusAlert = {
+  "annotations": {
+    [key: string]: string,
+  },
```

Comment:
OK for now, but we'll need to type some of the expected keys (e.g. `description`, `message`, ...). We've been hit by a missing key recently (see #2994).

Reply:
if they are always present in the alerts I can add typing for `description` and `summary`. I'm just unsure Prometheus alert manager will always send those keys as the documentation is not mentioning this.

Reply:
No they are optional indeed, so not sure if Flow can help in any way here.